lms/djangoapps/courseware/views/views.py

```diff
@@ -1171,12 +1171,19 @@ def _course_home_redirect_enabled():
 
 @login_required
 @ensure_valid_course_key
-def submission_history(request, course_id, student_username, location):
+def submission_history(request, course_id, student_username_or_email, location):
     """Render an HTML fragment (meant for inclusion elsewhere) that renders a
     history of all state changes made by this user for this problem location.
     Right now this only works for problems because that's all
     StudentModuleHistory records.
     """
+    try:
+        if '@' in student_username_or_email:
+            user = User.objects.get(email=student_username_or_email)
+        else:
+            user = User.objects.get(username=student_username_or_email)
```

Comment:
Touchstone collaborator accounts will have `@` in the username, which is allowed in edX when unicode usernames are turned on in settings.

You might want to look at how we handled this in similar situations, like https://github.com/mitodl/edx-platform/pull/70

Reply:
@pdpinch done. take another look.